sway-lib-core/src/codec.sw

```diff
@@ -22,12 +22,12 @@ impl AsRawSlice for Buffer {
 
 pub struct BufferReader {
     ptr: raw_ptr,
-    pos: u64,
+    pos: raw_ptr,
 }
```

Comment:
In the code changes it looks like `ptr` is always equal to `pos`. Is that just because you've only changed the cases where `pos = 0`?

Reply:
Ah sorry. That was intermediate code. I fixed this already.